certbot-apache/certbot_apache/_internal/override_centos.py

```diff
@@ -51,6 +52,38 @@ def config_test(self) -> None:
             else:
                 raise
 
+    def _rhel9_or_newer(self) -> bool:
+        os_name, os_version = util.get_os_info()
+        rhel_derived = os_name in [
+            "centos", "centos linux",
+            "cloudlinux",
+            "ol", "oracle",
+            "rhel", "redhatenterpriseserver", "red hat enterprise linux server",
+            "scientific", "scientific linux",
+        ]
+        at_least_v9 = util.parse_loose_version(os_version) >= util.parse_loose_version('9')
+        return rhel_derived and at_least_v9
+
+    def _override_cmds(self) -> None:
+        """
+        As of RHEL 9, apachectl can't be passed flags like "-v" or "-t -D", so instead
+        use options.bin (i.e. httpd) for version_cmd and the various get_X commands
+        """
+        if self._rhel9_or_newer():
+            if not self.options.bin:
+                raise ValueError("OS option apache_bin must be set for CentOS") # pragma: no cover
+
+            self.options.version_cmd[0] = self.options.bin
+            self.options.get_modules_cmd[0] = self.options.bin
+            self.options.get_includes_cmd[0] = self.options.bin
+            self.options.get_defines_cmd[0] = self.options.bin
+        else:
+            super()._override_cmds()
```

Comment:
I think we may want to unconditionally call this at the top of this function so values like `self.options.restart_cmd[0]` get set to a possibly custom `self.options.ctl` value. Otherwise, this PR LGTM!

Reply:
ack, yup!